core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -833,8 +840,9 @@ private void checkProfilesConsistency() {
                         "\nYou need to add `|turn_costs=true` to the vehicle in `graph.flag_encoders`");
             }
             try {
-                createWeighting(new HintsMap(profile.getWeighting()), encoder, NO_TURN_COST_PROVIDER);
+                createWeighting(profile, new PMap());
             } catch (IllegalArgumentException e) {
+                // todonow: update, what are we really checking here?
                 throw new IllegalArgumentException("The profile '" + profile.getName() + "' was configured with an unknown weighting '" + profile.getWeighting() + "'");
```

Comment:
In #1776 I stumbled over this check as I wanted to inject the graph into the Weighting but at this time graph is null. 

But with #1950 I found a good solution so that I kept this method here :)

Reply:
Ah ok. The graph being null problem when running this check should also be already fixed with this: 639bd9ab69ae63cd0ea580e9076b9c1679e855b5

Reply:
Aha, where did you stumble over this?

Reply:
Well its required for this branch because the turn cost provider is created in `createWeighting` now (and thus there is a call to `ghStorage.getTurnCostStorage()`).